### PR TITLE
Use disabledColor instead of Grey[500] for disabled button

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

File: step_2/lib/main.dart [hunk 1]
```diff
@@ -71,8 +71,8 @@ class ChatScreenState extends State<ChatScreen> {
               margin: const EdgeDims.symmetric(horizontal: 4.0),
               child: new FloatingActionButton(
                 child: new Icon(icon: 'content/send', size: IconSize.s18),
-                onPressed: () => _handleMessageAdded(_currentMessage),
-                backgroundColor: _isComposing ? null : Colors.grey[500],
+                onPressed: _isComposing ? () => _handleMessageAdded(_currentMessage) : null,
+                backgroundColor: _isComposing ? null : Theme.of(context).disabledColor,
```
Comment (Hixie) [inline step_2/lib/main.dart]: why do you need the background color at all? the button should take care of it.

**Comment by Hixie** [comment]
lgtm

File: step_2/lib/main.dart (hunk 1 above)
Comment (collinjackson) [inline step_2/lib/main.dart]: I tested on the latest master and setting onPressed to null doesn't set the color to disabled automatically. Should it?

Comment (Hixie) [inline step_2/lib/main.dart]: Oh, floating action button might not support that.

Actually, having a disabled floating action button might be non-material. @HansMuller do you have any advice here?

Comment (collinjackson) [inline step_2/lib/main.dart]: image

image

I was basing it on the Google Hangouts app which looks like this. I'm not sure if it's a small floating action button or a circular raised button.

Comment (Hixie) [inline step_2/lib/main.dart]: Wow yeah, I don't even know what that is.

Anyway I guess we should make FABs support this just like the other buttons. But until we do that, this is LGTM.

Comment (HansMuller) [inline step_2/lib/main.dart]: I've tacked the disabled feature on to https://github.com/flutter/flutter/issues/227

**Pull request merged**